packages/editor/src/editor.ts

```diff
@@ -0,0 +1,207 @@
+
+    // Check for an external editor in the environment variable.
+    if (!editor && process.env.EDITOR) {
+      editor = process.env.EDITOR;
+    }
+
+    return editor;
+  }
+
+  async _createTempFile({ content, ext }: { content: string; ext: string }): Promise<string> {
+    const tmpDoc = path.join(this._tmpDir, `edit-${new bson.ObjectId()}.${ext}`);
+
+    // Create a temp file to store content that is being edited.
+    await fs.mkdir(path.dirname(tmpDoc), { recursive: true, mode: 0o700 });
+    await fs.writeFile(tmpDoc, beautify(content));
+
+    return tmpDoc;
+  }
+
+  async _readTempFile(tmpDoc: string): Promise<string> {
+    const content = await fs.readFile(tmpDoc, 'utf8');
+    await fs.unlink(tmpDoc);
+    return this._makeMultilineJSIntoSingleLine(content);
+  }
+
+  _isVscodeApp(cmd: string): boolean {
+    const regex = /^(.*)[\/\\]?[cC]ode(.exe)?$/;
+    return regex.test(cmd);
+  }
+
+  _isIdentifier(code: string): boolean {
+    const regex = /^([^!#%&()*+,\-/\\^`{|}~]+)$/;
+    return regex.test(code);
+  }
+
+  async _getEditorContent(code: string): Promise<string> {
+    if (!code) {
+      return this._content;
+    }
+
+    if (!this._isIdentifier(code)) {
+      return code;
+    }
+
+    const evalResult = await this._loadExternalCode(code, '@(shell eval)');
+    return evalResult.toString();
+  }
+
+  async runEditCommand(args: string[]): Promise<void> {
+    await this.print('Opening an editor...');
+
+    const code = args.join(' ');
+    const editor: string|null = await this._getEditor();
+
+    // If none of the above configurations are found return an error.
+    if (!editor) {
+      throw new Error('Command failed with an error: please define an external editor');
+    }
+
+    const content = await this._getEditorContent(code);
+    const ext = await this._getExtension(editor);
+    const tmpDoc = await this._createTempFile({ content, ext });
+
+    this.messageBus.emit('mongosh-editor:run-edit-command', {
+      tmpDoc,
+      editor,
+      code
+    });
+
+    const proc = spawn(editor, [tmpDoc], { stdio: 'inherit' });
+    let stdout = '';
+    let stderr = '';
+
+    if (proc.stdout) {
+      (proc.stdout as NodeJS.ReadableStream).setEncoding('utf8').on('data', (chunk) => { stdout += chunk; });
+    }
+    if (proc.stderr) {
+      (proc.stdout as NodeJS.ReadableStream).setEncoding('utf8').on('data', (chunk) => { stderr += chunk; });
+    }
+
+    this._input.pause();
+
+    try {
+      const [ exitCode ] = await once(proc, 'exit');
+
+      if (exitCode === 0) {
+        const content = await this._readTempFile(tmpDoc);
+        this._input.unshift(content);
+        return;
+      }
+
+      // Allow exit code 1 if stderr is empty, i.e. no error occurred, because
+      // that is how commands like `npm outdated` report their result.
+      if (exitCode === 1 && stderr === '' && stdout) {
+        stderr = stdout;
+      }
+
```

Comment:
```suggestion
```

since this was specifically about `npm outdated`, and we’re not running that here :)